lib/src/server/server.dart

```diff
@@ -320,14 +334,11 @@ class Server extends ConnectionServer {
   }
 
   Future<void> shutdown() async {
-    final done = _connections.map((connection) => connection.finish()).toList();
-    if (_insecureServer != null) {
-      done.add(_insecureServer!.close());
-    }
-    if (_secureServer != null) {
-      done.add(_secureServer!.close());
-    }
-    await Future.wait(done);
+    await Future.wait([
+      ..._connections.map((connection) => connection.finish()),
```

Comment:
Consider instead of `...` and `map` just:

```
await Future.wait([
  for (var connection in _connections) connection.finish(),
  // ...
]);
```

Reply:
Nice, I didn't know that worked.